independent-projects/bootstrap/app-model/src/main/java/io/quarkus/paths/PathVisit.java

```diff
@@ -4,12 +4,34 @@
 import java.net.URL;
 import java.nio.file.Path;
 
+/**
+ * Provides context for a given path visit
+ */
 public interface PathVisit {
 
+    /**
+     * The root of the path tree the current path belongs to.
+     * For a {@link PathTree} created for an archive, this will be the path to the archive file.
+     * For a {@link PathTree} created for a directory, this will be the path to the directory.
```

Comment:
Absolute or relative (to what)?

Reply:
It's the caller that provides the value of the root.

Reply:
For both `ArchivePathTree` and `DirectoryPathTree` you seem to be calling `Files.walk(rootDir)` right after creating PathTreeVisit so the value of rootDir must be de facto absolute. I guess the API users are nor supposed to add new implementations of PathVisit and they mostly interact with it via their custom PathVisitors that they pass to PathTree.walk(PathVisitor). So for the API users, PathVisit instances are supplied by the API and they deserve to know whether the paths returned by PathVisit methods are (de facto) absolute.

Reply:
> you seem to be calling Files.walk(rootDir) right after creating PathTreeVisit so the value of rootDir must be de facto absolute

The `rootDir` does not have to be an absolute path for `Files.walk(rootDir)` to work though.

Reply:
Peter, based on my comment above, I wouldn't say it's "de facto absolute". It really depends on what the `rootDir` path is, which is provided by the caller.
Otherwise, from perspective, this javadoc is pretty much in line with https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/nio/file/FileVisitor.html Or am I still missing something? Or you'd also have similar remarks for that the `FileVisitor` javadoc?

Reply:
> Peter, based on my comment above, I wouldn't say it's "de facto absolute". It really depends on what the `rootDir` path is, which is provided by the caller.

Sorry, "de facto absolute" is perhaps misleading. What I mean is "ready for IO operations" or simply "granted to exist". I admit it is rather trivial to document for getRoot() but it would be really nice to add a sentence to getPath() stating something like

> The returned path is granted to exist in the underlying filesystem (unless it was deleted by a parallel process or thread) and therefore it can be used for I/O operations straight away without further resolving e.g. against getRoot().